src/test/shared/sam/debugger/samDebugConfigProvider.test.ts

```diff
@@ -206,6 +218,39 @@ describe('SamDebugConfigurationProvider', async () => {
                 runtime: 'happy-runtime-42',
             }
             assert.deepStrictEqual(await debugConfigProvider.makeConfig(config.folder, config.config), undefined)
+
+            // bad credentials
+            const mockCredentialsStore: CredentialsStore = new CredentialsStore()
+
+            const credentialsProvider: CredentialsProvider = {
+                getCredentials: sandbox.stub().resolves(({} as any) as AWS.Credentials),
+                getCredentialsProviderId: sandbox.stub().returns({
+                    credentialType: 'test',
+                    credentialTypeId: 'someId',
+                }),
+                getDefaultRegion: sandbox.stub().returns('someRegion'),
+                getHashCode: sandbox.stub().returns('1234'),
+                canAutoConnect: sandbox.stub().returns(true),
+            }
+            const getCredentialsProviderStub = sandbox.stub(
+                CredentialsProviderManager.getInstance(),
+                'getCredentialsProvider'
+            )
+            getCredentialsProviderStub.resolves(credentialsProvider)
+            sandbox.stub(mockCredentialsStore, 'upsertCredentials').throws()
+            const debugConfigProviderMockCredentials = new SamDebugConfigProvider({
+                ...fakeContext,
+                credentialsStore: mockCredentialsStore,
+            })
+            assert.deepStrictEqual(
+                await debugConfigProviderMockCredentials.makeConfig(config.folder, {
+                    ...config.config,
+                    aws: {
+                        credentials: 'profile:error',
+                    },
+                }),
+                undefined
```

Comment:
is all the mocking really needed just to get a failure here from an input of `profile:error`...?